airflow/utils/log/file_task_handler.py

```diff
@@ -194,10 +195,7 @@ def _init_file(self, ti):
         # operator is not compatible with impersonation (e.g. if a Celery executor is used
         # for a SubDag operator and the SubDag operator has a different owner than the
         # parent DAG)
-        if not os.path.exists(directory):
-            # Create the directory as globally writable using custom mkdirs
-            # as os.makedirs doesn't set mode properly.
-            mkdirs(directory, 0o777)
+        pathlib.Path(directory).mkdir(mode=0o777, parents=True, exist_ok=True)
```

Comment:
We should really stop the 777 madness

Reply:
Yes, let's do 644? I'm not sure what the implications are but 777 is definitely wrong.

Reply:
The comment above has quite detailed explanation and it seems in order to stop the madness we would have to fix impersonation for SubDag oprerator with different owner in case of re-runs... It's pretty edge case, but I guess it's quite valid (though not secure of course).

Reply:
How did this ever get in here 😞 Even though it's a valid comment, 777 is never the correct solution.

Seems like this will take some time to test and fix...